priiloader/source/main.cpp

```diff
@@ -2461,7 +2463,49 @@ int main(int argc, char **argv)
 	GetStateFlags(&flags);
 	gprintf("Bootstate %u detected. DiscState %u ,ReturnTo %u & Flags %u & checksum %u (gcflag : 0x%08X)", flags.type, flags.discstate, flags.returnto, flags.flags, flags.checksum, GcShutdownFlag);
 	s8 magicWord = CheckMagicWords();
-	
+	bool forcePriiloaderMenu = false;
+
+	if (CheckvWii())
+	{
+		ImportWiiUConfig();
+		const WiiUConfig* wiiuConfig = GetWiiUConfig();
+
+		// Read AR from Wii U settings if it exists
+		s32 ar;
+		if (wiiuConfig)
+			ar = wiiuConfig->av.ipl_ar;
+		else
+			ar = CONF_GetAspectRatio();
+
+		// Now set the correct DMCU aspect ratio
+		STM_DMCUWrite(ar ? 0x30000004 : 0x10000002);
+
+		// bit of a hack but BC-NAND clears the boot state, so let's try to guess it
+		if (!IsInitialBoot())
+			Bootstate = TYPE_RETURN;
+		else
+			Bootstate = TYPE_UNKNOWN;
+
+		flags.type = Bootstate;
+		SetBootState(flags.type, flags.flags, flags.returnto, flags.discstate);
+
+		// Check if Wii U side wants to open the Priiloader menu
+		const WiiUArgs* wiiuArgs = GetWiiUArgs();
+		if (wiiuArgs)

```

Comment:
these ifs can be merged into 1 `if(wiiuArgs != null && (wiiuArgs->flags & 0x10) ...`. i don't like nested if's, ive become allergic to them :sweat_smile: